File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/models/Origin.java
package io.appium.espressoserver.lib.helpers.w3c.models;

import com.google.gson.annotations.JsonAdapter;

import javax.annotation.Nullable;

import static io.appium.espressoserver.lib.helpers.w3c.models.InputSource.VIEWPORT;

@JsonAdapter(OriginDeserializer.class)
public class Origin {
    private String type = VIEWPORT;
    private String elementId;

    public Origin() {
    }

    public Origin(String type) {
        this.type = type;
    }

    @Nullable
    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getElementId() {
        return elementId;
    }

    public void setElementId(String elementId) {
        this.elementId = elementId;
    }
}

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/models/Origin.kt
package io.appium.espressoserver.lib.helpers.w3c.models

import com.google.gson.annotations.JsonAdapter

@JsonAdapter(OriginDeserializer::class)
class Origin {
    var type: String? = InputSource.VIEWPORT
    var elementId: String? = null

    constructor() {}
    constructor(type: String) {
        this.type = type
    }
}